File: src/Talabi.Core/Entities/Product.cs
using Talabi.Core.Enums;

namespace Talabi.Core.Entities;

public class Product : BaseEntity
{
    public Guid VendorId { get; set; }
    public Vendor? Vendor { get; set; }
    public VendorType? VendorType { get; set; }

    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Category { get; set; } // Deprecated, use CategoryId
    public Guid? CategoryId { get; set; }
    public Category? ProductCategory { get; set; }
    public decimal Price { get; set; }
    public Currency Currency { get; set; } = Currency.TRY; // Required currency
    public string? ImageUrl { get; set; }

    // Product availability and stock management
    public bool IsAvailable { get; set; } = true;
    public int? Stock { get; set; }
    public int? PreparationTime { get; set; } // in minutes

    public ICollection<ProductOptionGroup> OptionGroups { get; set; } = new List<ProductOptionGroup>();
}

File: mobile/lib/models/product.dart
class Product {
  final String id;
  final String vendorId;
  final String? vendorName;
  final String name;
  final String? description;
  final String? category;
  final String? categoryId;
  final double price;
  final String? imageUrl;
  final bool isAvailable;
  final int? stock;
  final int? preparationTime;

  Product({
    required this.id,
    required this.vendorId,
    this.vendorName,
    required this.name,
    this.description,
    this.category,
    this.categoryId,
    required this.price,
    this.imageUrl,
    this.isAvailable = true,
    this.stock,
    this.preparationTime,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'].toString(),
      vendorId: json['vendorId'].toString(),
      vendorName: json['vendorName'],
      name: json['name'],
      description: json['description'],
      category: json['category'],
      categoryId: json['categoryId']?.toString(),
      price: (json['price'] as num).toDouble(),
      imageUrl: json['imageUrl'],
      isAvailable: json['isAvailable'] ?? true,
      stock: json['stock'],
      preparationTime: json['preparationTime'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'vendorId': vendorId,
      'vendorName': vendorName,
      'name': name,
      'description': description,
      'category': category,
      'categoryId': categoryId,
      'price': price,
      'imageUrl': imageUrl,
      'isAvailable': isAvailable,
      'stock': stock,
      'preparationTime': preparationTime,
    };
  }
}
